table.go

```diff
@@ -69,17 +65,25 @@ func NumDigits(b *big.Int) int64 {
 	if bl == 0 {
 		return 1
 	}
-	if val, ok := lookupBits(bl); ok {
-		ab := new(big.Int).Abs(b)
-		if ab.Cmp(&val.border) < 0 {
+
+	if bl < digitsTableLen {
+		val := digitsLookupTable[bl]
+		a := b
+		if b.Sign() < 0 {
+			a = new(big.Int).Abs(b)
+		}
+		if a.Cmp(&val.border) < 0 {
 			return val.digits
 		}
 		return val.digits + 1
 	}
 
 	n := int64(float64(bl) / digitsToBitsRatio)
-	a := new(big.Int).Abs(b)
-	e := new(big.Int).Exp(bigTen, big.NewInt(n), nil)
+	e := tableExp10(n, nil)
+	a := b
+	if b.Sign() < 0 {
+		a = new(big.Int).Abs(b)
```

Comment:
Done.

Since tableExp10 may return a value from a lookup table, it should not be mutated. I'm unclear why this is unclear. Can you suggest better wording? For example, this would be wrong:

```
a := tableExp10(5, nil)
a.Add(a, a)
```

because it would modify the big.Int in the lookup table.